msdm/algorithms/search.py

```diff
@@ -95,49 +99,47 @@ def __init__(
         heuristic_value=lambda s: 0,
         seed=None,
         randomize_action_order=False,
-        # tie_breaking_strategy="random",
     ):
         self.heuristic_value = heuristic_value
         self.seed = seed
-        # self.tie_breaking_strategy = tie_breaking_strategy
         self.randomize_action_order = randomize_action_order
 
-    def plan_on(self, dss: DeterministicShortestPathProblem):
+    def plan_on(self, dsp: DeterministicShortestPathProblem):
         rnd = random.Random(self.seed)
         if self.randomize_action_order:
             shuffled = make_shuffled(rnd)
         else:
             shuffled = lambda list: list
         
-        dss = DeterministicShortestPathProblem.from_mdp(dss)
+        dsp = DeterministicShortestPathProblem.from_mdp(dsp)
 
         # Every queue entry is a pair of
         # - a tuple of priorities/costs (the cost-to-go, cost-so-far, and a random tie-breaker)
         # - the state
         queue = []
-        start = dss.initial_state()
+        start = dsp.initial_state()
         heapq.heappush(queue, ((-self.heuristic_value(start), 0, rnd.random()), start))
```

Comment:
entirely unrelated, but I do wonder if we should swap out the `rnd.random()` here for a counter, to implement LIFO/FIFO dynamics? I think you can get LIFO with a counter that is decremented for every insertion (here's [one of my implementations](https://github.com/cgc/resource-rational-task-decomposition/blob/main/rrtd/search_algorithms.py#L117-L121)).